pkg/controller/monitor/monitor_controller.go

```diff
@@ -191,6 +199,45 @@ func (r *ReconcileMonitor) Reconcile(ctx context.Context, request reconcile.Requ
 		}
 	}
 
+	var tlsSecret *corev1.Secret
+	if install.CertificateManagement == nil {
+		// Check that if the apiserver cert pair secret exists that it is valid (has key and cert fields)
+		// If it does not exist then this function still returns true
+		tlsSecret, err = utils.ValidateCertPair(r.client,
+			common.OperatorNamespace(),
+			monitor.PrometheusTLSSecretName,
+			corev1.TLSPrivateKeyKey,
+			corev1.TLSCertKey,
+		)
+		if err != nil {
+			log.Error(err, "Invalid TLS Cert")
+			r.status.SetDegraded("Error validating TLS certificate", err.Error())
+			return reconcile.Result{}, err
+		}
+		r.status.RemoveCertificateSigningRequests(common.TigeraPrometheusNamespace)
+	} else {
+		// Monitor pending CSRs for the TigeraStatus
+		r.status.AddCertificateSigningRequests(common.TigeraPrometheusNamespace, map[string]string{"k8s-app": common.TigeraPrometheusNamespace})
+	}
+
+	// Fetch the Authentication spec. If present, we use to configure user authentication.
+	authenticationCR, err := utils.GetAuthentication(ctx, r.client)
```

Comment:
Just for my understanding, we're adding OIDC login capabilities to the prometheus UI, which is why we need this configuration, right?

Reply:
Yes, it is for reading the bearer tokens. Although, with the current implementation, it will mostly read tigera-manager impersonation tokens.